htdocs/accountancy/journal/purchasesjournal.php

```diff
@@ -214,9 +215,10 @@
 					$bookkeeping->doc_type = 'supplier_invoice';
 					$bookkeeping->fk_doc = $key;
 					$bookkeeping->fk_docdet = 0;    // Useless, can be several lines that are source of this record to add
-					$bookkeeping->code_tiers = $tabcompany[$key]['code_fournisseur'];
-					$bookkeeping->label_compte = dol_trunc($companystatic->name, 16) . ' - ' . $invoicestatic->refsupplier . ' - ' . $langs->trans("Code_tiers");
-					$bookkeeping->numero_compte = $tabcompany[$key]['code_compta_fournisseur'];
+					$bookkeeping->code_tiers = $tabcompany[$key]['code_compta_fournisseur'];
```

Comment:
code_tiers is the code of the thirdparty, not the accounting account.
accounting account must be store into numero_compte